ledger/acctupdates.go

```diff
@@ -920,6 +927,16 @@ func (aul *accountUpdatesLedgerEvaluator) CheckDup(config.ConsensusParams, basic
 	return fmt.Errorf("accountUpdatesLedgerEvaluator: tried to check for dup during accountUpdates initialization ")
 }
 
+// BlockTimeStamp gets the block timestamp of a specific round.
+func (aul *accountUpdatesLedgerEvaluator) BlockTimeStamp(rnd basics.Round) int64 {
+	if ts, ok := aul.blockTimeStamps[rnd]; ok {
+		return ts
+	}
+	hdr, _ := aul.au.ledger.BlockHdr(rnd)
```

Comment:
it's ok that you don't want to test the error here... but you'll need to preempt that test and actively verify that the blocks database truly contain <first round> - 1001. This is part of the startup code, so we need to test inconsistencies between the two databases.

Reply:
make sense - will add a pre-check and test it

Reply:
this would also be a good place to potentially delete old entries, in case we're catching up more than 1000 rounds.
i.e.
```golang
delete(aul.blockTimeStamps, rnd - 1001)
```

Reply:
this actually ended up to be changes in `au.committedUpTo`: when blockdb has 1005 rounds, `txTail` says `4 = 1005 - 1001` and `au` says something like `1005 - 320 - X` where X relatively small number of not committed yet offsets. But in fact block 999 might have `txn FirstValidTime` with `FirstValid=1` (or 0), and blockdb does not have such round. That means for any round R: `au.dbRound < R <= Latest` blockdb must have a block for round `R - (MaxTxnLife + 1)`. This is 320 + X rounds more than `txTail` :(

```
               au.dbRound                       Latest
                     |  X  |    Lookback (320)    |
---------------------+-----+----------------------+
```

Reply:
Added test and checking code